packages/react-arborist/src/types.ts

```diff
@@ -1,148 +0,0 @@
-import React, {
-  ComponentType,
-  CSSProperties,
-  MouseEvent,
-  MouseEventHandler,
-  MutableRefObject,
-  ReactElement,
-  ReactNode,
-  Ref,
-} from "react";
-import { FixedSizeList } from "react-window";
-import { Cursor } from "./dnd/compute-drop";
-import { SelectionData } from "./selection/selection";
-import { TreeApi } from "./tree-api";
-
-// Forward ref can't forward generics without this little re-declare
-// https://fettblog.eu/typescript-react-generic-forward-refs/
-declare module "react" {
-  function forwardRef<T, P = {}>(
-    render: (props: P, ref: React.Ref<T>) => React.ReactElement | null
-  ): (props: P & React.RefAttributes<T>) => React.ReactElement | null;
-}
-
-export type Node<T = unknown> = {
-  id: string;
-  model: T;
-  level: number;
-  children: Node<T>[] | null;
-  parent: Node<T> | null;
-  isOpen: boolean;
-  isDraggable: boolean;
-  isDroppable: boolean;
-  rowIndex: number | null;
-};
-
-export type NodesById<T> = { [id: string]: Node<T> };
-
-export interface IdObj {
-  id: string;
-}
-
-export type NodeRendererProps<T extends IdObj> = {
-  innerRef: (el: HTMLDivElement | null) => void;
-  styles: { row: CSSProperties; indent: CSSProperties };
-  data: T;
-  state: NodeState;
-  handlers: NodeHandlers;
-  tree: TreeApi<T>;
-  preview: boolean;
-};
-
-export type NodeState = {
-  isOpen: boolean;
-  isSelected: boolean;
-  isHoveringOverChild: boolean;
-  isDragging: boolean;
-  isSelectedStart: boolean;
-  isSelectedEnd: boolean;
-  isEditing: boolean;
-};
-
-export type NodeHandlers = {
```

Comment:
When I tried this rc branch with my project, I was getting an error where the type `NodeHandlers` is missing. I saw that it was still used in the demo code for this branch and I am wondering what is the alternative type.

Reply:
@shorttermproject There is no more NodeHandlers. There is now a NodeApi instance. More docs to come, but here's how it works now in some code:

https://github.com/brimdata/react-arborist/blob/version-2/packages/showcase/pages/simple-tree/index.tsx#L16-L34

Reply:
@shorttermproject I've updated the README in this branch with examples and the API reference. It should be complete enough to get going.